app/controlplane/internal/biz/signing.go

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package biz
+
+import (
+	"context"
+	"crypto"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"errors"
+
+	"github.com/sigstore/fulcio/pkg/ca"
+	"github.com/sigstore/fulcio/pkg/identity"
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
+)
+
+type SigningCertCreator interface {
+	// CreateSigningCert creates a signing certificate from and returns the certificate chain
+	CreateSigningCert(context.Context, string, []byte) ([]string, error)
```

Comment:
what's this interface for? Note that in general we are not creating interfaces in the use-cases. The interfaces that we have there are on the `repo` layer.

Reply:
I was assuming that we wanted to have interfaces at this level as well (to create test mocks, etc.). But maybe it's something we can do for the whole layer in a future change. I'll keep it simple for now.